### ISSUE TITLE
Add tests for `config` command and improve overall robustness

### ISSUE THREAD

**Issue by jefflester** [jefflester/minitrino#86]
A simple case can produce a very annoying error. Start with this config file with an invalid section name:

```
[configgg]
# defaults to ~/.minitrino/lib
LIB_PATH=

STARBURST_VER=
TEXT_EDITOR=code
LIC_PATH=
```

Then:

```
minitrino -v config
[e]  Failed to parse config file: /Users/jlester/.minitrino/minitrino.cfg
[e]  No section: 'config'

Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.10/3.10.13/Frameworks/Python.framework/Versions/3.10/lib/python3.10/configparser.py", line 848, in items
    d.update(self._sections[section])
KeyError: 'config'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/jlester/work/repos/minitrino/src/cli/minitrino/components.py", line 260, in _parse_minitrino_config
    for k, v in config.items("config"):
  File "/opt/homebrew/Cellar/python@3.10/3.10.13/Frameworks/Python.framework/Versions/3.10/lib/python3.10/configparser.py", line 851, in items
    raise NoSectionError(section)
configparser.NoSectionError: No section: 'config'
```

And even `--reset` doesn't work:

```
minitrino -v config --reset
[e]  Failed to parse config file: /Users/jlester/.minitrino/minitrino.cfg
[e]  No section: 'config'

Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.10/3.10.13/Frameworks/Python.framework/Versions/3.10/lib/python3.10/configparser.py", line 848, in items
    d.update(self._sections[section])
KeyError: 'config'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/jlester/work/repos/minitrino/src/cli/minitrino/components.py", line 260, in _parse_minitrino_config
    for k, v in config.items("config"):
  File "/opt/homebrew/Cellar/python@3.10/3.10.13/Frameworks/Python.framework/Versions/3.10/lib/python3.10/configparser.py", line 851, in items
    raise NoSectionError(section)
configparser.NoSectionError: No section: 'config'
```

Add a few tests to the CLI testing library and change the code as necessary to account for the above scenarios.

**Comment by jefflester** [comment]
Resolved by 254ff64be9c214e006a9dfa5c6c6a38e68ca9c9b and ab96e37060939c58507cb43da7a955b60b6dceb7. Will close this when PR is merged

**Comment by jefflester** [comment]
Closed with #87
